src/main/flight/mixer.c

```diff
@@ -424,42 +425,72 @@ static void updateDynLpfCutoffs(timeUs_t currentTimeUs, float throttle)
-    const float motorMixDelta = 0.5f * motorMixRange;
+    float motorMixNormalizationFactor;
+    float motorDeltaScale;
+    if (airmodeEnabled || throttle > 0.5f) {
+        motorMixNormalizationFactor = motorMixRange > 1.0f ? 1.0f / motorMixRange : 1.0f;
+        motorDeltaScale = 0.5f;
+    } else {
+        // this scales the motor mix authority to be 0.5 at 0 throttle, and 1.0 at 0.5 throttle as airmode off intended for things to work.
+        // also lays the groundwork for how an airmode percent would work.
+        const float airmodeTransitionPercent = scaleRangef(throttle, 0.0f, 0.5f, 0.5f, 1.0f); // 0.5 throttle is full transition, and 0.0 throttle is 50% airmodeTransitionPercent
+        motorMixNormalizationFactor = motorMixRange > airmodeTransitionPercent ? airmodeTransitionPercent / motorMixRange : 1.0f;
+        motorDeltaScale = 0.5f * airmodeTransitionPercent; // this should be half of the motor authority allowed
+    }
 
+    const float motorMixDelta = motorDeltaScale * motorMixRange;
+
+    float minMotor = FLT_MAX;
+    float maxMotor = FLT_MIN;
     for (int i = 0; i < mixerRuntime.motorCount; ++i) {
-        if (airmodeEnabled || throttle > 0.5f) {
-            if (mixerConfig()->mixer_type == MIXER_LINEAR) {
-                motorMix[i] = scaleRangef(throttle, 0.0f, 1.0f, motorMix[i] + motorMixDelta, motorMix[i] - motorMixDelta);
-            } else {
-                motorMix[i] = scaleRangef(throttle, 0.0f, 1.0f, motorMix[i] + ABS(motorMix[i]), motorMix[i] - ABS(motorMix[i]));
-            }
+        if (mixerConfig()->mixer_type == MIXER_LINEAR) {
+            motorMix[i] = scaleRangef(throttle, 0.0f, 1.0f, motorMix[i] + motorMixDelta, motorMix[i] - motorMixDelta);
+        } else {
+            motorMix[i] = scaleRangef(throttle, 0.0f, 1.0f, motorMix[i] + ABS(motorMix[i]), motorMix[i] - ABS(motorMix[i]));
+        }
+
+        maxMotor = MAX(motorMix[i], maxMotor);
+
+        minMotor = MIN(motorMix[i], minMotor);
+
+        // correct throttle so it won't clip any outputs
+        if (minMotor + throttle < 0.0f) {
+            throttle = -minMotor * motorMixNormalizationFactor;
+        } else if (maxMotor + throttle > 1.0f) {
+            throttle = 1.0f - maxMotor * motorMixNormalizationFactor;
         }
-        motorMix[i] /= motorMixNormalizationFactor;
+
+        motorMix[i] *= motorMixNormalizationFactor;
     }
 }
 
 static void applyMixerAdjustment(float *motorMix, const float motorMixMin, const float motorMixMax, const bool airmodeEnabled) {
 #ifdef USE_AIRMODE_LPF
     const float unadjustedThrottle = throttle;
     throttle += pidGetAirmodeThrottleOffset();
-    float airmodeThrottleChange = 0;
+    float airmodeThrottleChange = 0.0f;
 #endif
-    
-    const float motorMixNormalizationFactor = motorMixRange > 1.0f ? motorMixRange : 1.0f;
+    float motorMixNormalizationFactor;
+    if (airmodeEnabled || throttle > 0.5f) {
```

Comment:
newline before next block please.

Reply:
Sorry for lame review - only space issues for coding readability - not flight related

Reply:
fixed spacing